src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp

```diff
@@ -2922,6 +2922,56 @@ struct AddSimplify : public OpRewritePattern<mlir::stablehlo::AddOp> {
   }
 };
 
+// ((add x cst0) cst1) -> (add x1 (add cst0 cst1))
+template <typename T> struct BinOpConstSimplify : public OpRewritePattern<T> {
+  using OpRewritePattern<T>::OpRewritePattern;
```

Comment:
you should template this with a fastmath flag, since while this is always valid for ints, for floats this may violate IEEE standards

Reply:
I applied only to integers for now.

https://github.com/EnzymeAD/Enzyme-JAX/blob/06eb601b6518769ef4ea8760433726adcd7f619b/src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp#L2931-L2933

Reply:
Fair nuff, open an issue for extending to floats?